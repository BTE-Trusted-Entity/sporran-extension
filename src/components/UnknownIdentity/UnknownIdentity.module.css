.container {
  composes: container from '../View/View.module.css';
}

.heading {
  composes: screenHeadline from '../../components/Typography/Typography.module.css';
  margin-bottom: 2rem;
}

.text {
  composes: subline from '../../components/Typography/Typography.module.css';
}
