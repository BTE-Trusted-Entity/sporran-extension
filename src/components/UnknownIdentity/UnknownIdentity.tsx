import type { KiltAddress } from '@kiltprotocol/sdk-js';

import { browser } from 'webextension-polyfill-ts';

import * as styles from './UnknownIdentity.module.css';

export function UnknownIdentity({
  address,
}: {
  address: KiltAddress;
}): JSX.Element {
  const t = browser.i18n.getMessage;
  return (
    <section className={styles.container}>
      <h1 className={styles.heading}>Error</h1>
      <p className={styles.text}>
        {t('component_UnknownIdentity_text', [address])}
      </p>
      <p className={styles.text}>{t('component_UnknownIdentity_info')}</p>
    </section>
  );
}
