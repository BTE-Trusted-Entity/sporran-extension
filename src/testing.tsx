export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
