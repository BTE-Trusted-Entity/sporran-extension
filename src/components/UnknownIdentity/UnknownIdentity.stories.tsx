import { Meta } from '@storybook/react';

import { UnknownIdentity } from './UnknownIdentity';

export default {
  title: 'Components/UnknownIdentity',
  component: UnknownIdentity,
} as Meta;

export function Template(): JSX.Element {
  return (
    <UnknownIdentity address="4qp7KB8jbwqS6XXL8zH8qZn3GCdnZDt6Nmq5M47ztGKhXJVh" />
  );
}
