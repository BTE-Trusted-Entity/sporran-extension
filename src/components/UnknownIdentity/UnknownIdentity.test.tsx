import { render } from '../../testing/testing';

import { UnknownIdentity } from './UnknownIdentity';

describe('UnknownIdentity', () => {
  it('should match the snapshot', async () => {
    const { container } = render(
      <UnknownIdentity address="4qp7KB8jbwqS6XXL8zH8qZn3GCdnZDt6Nmq5M47ztGKhXJVh" />,
    );
    expect(container).toMatchSnapshot();
  });
});
