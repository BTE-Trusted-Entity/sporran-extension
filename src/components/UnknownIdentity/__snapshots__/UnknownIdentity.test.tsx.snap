// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownIdentity should match the snapshot 1`] = `
<div>
  <section
    class="container"
  >
    <h1
      class="heading"
    >
      Error
    </h1>
    <p
      class="text"
    >
      The identity 4qp7KB8jbwqS6XXL8zH8qZn3GCdnZDt6Nmq5M47ztGKhXJVh is not recognized by Sporran.
    </p>
    <p
      class="text"
    >
      This operation is only supported for identities existing in Sporran.
    </p>
  </section>
</div>
`;
